packages/amplify-cli-core/src/amplify-node-pkg-detector/amplify-nodejs-detector.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { AmplifyError } from '../errors/amplify-error';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { getPackageManager, PackageManager } from '../utils/packageManager';
+import { Lockfile, LockfileParser } from './lock-file-interface';
+import { PackageJson } from './lock-file-types';
+import { LockFileParserFactory } from './parser-factory';
+
+/**
+ * return type of detectAffectedDirectDependencies
+ */
+export type DetectedDependencies = {
```

Comment:
Should this be DetectedDependency since it contains a singular package and the array of it would be dependencies?